server/repository/src/db_diesel/temperature_chart/temperature_chart.rs

```diff
@@ -0,0 +1,355 @@
+
+        let chart_data = query
+            .load::<QueryResult>(&self.connection.connection)?
+            .into_iter()
+            .map(TemperatureChartRow::from)
+            .collect();
+
+        Ok((chart_data, intervals))
+    }
+}
+
+impl From<QueryResult> for TemperatureChartRow {
+    fn from((from_datetime, to_datetime, average_temperature, sensor_id): QueryResult) -> Self {
+        Self {
+            from_datetime,
+            to_datetime,
+            average_temperature,
+            sensor_id,
+        }
+    }
+}
+
+fn calculate_intervals(
+    from_datetime: NaiveDateTime,
+    to_datetime: NaiveDateTime,
+    number_of_data_points: i32,
+) -> Vec<Interval> {
+    let interval = (to_datetime - from_datetime) / number_of_data_points;
+    (0..number_of_data_points)
+        .map(|point| Interval {
+            from_datetime: datetime_with_offset(&from_datetime, interval * point),
+            to_datetime: datetime_with_offset(&from_datetime, interval * (point + 1)),
+        })
+        .collect()
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{
+        location::LocationFilter,
+        mock::{MockData, MockDataInserts},
+        test_db::setup_all_with_data,
+        EqualFilter, LocationRow, NameRow, SensorFilter, SensorRow, StoreRow,
+        TemperatureChartRepository, TemperatureChartRow, TemperatureLogRow,
+    };
+    use util::create_datetime;
+    #[test]
+    fn test_calculate_intervals() {
+        assert_eq!(
+            calculate_intervals(
+                create_datetime(2021, 01, 01, 23, 59, 50).unwrap(),
+                create_datetime(2021, 01, 02, 00, 00, 20).unwrap(),
+                5
+            ),
+            vec![
+                Interval {
+                    from_datetime: create_datetime(2021, 01, 01, 23, 59, 50).unwrap(),
+                    to_datetime: create_datetime(2021, 01, 01, 23, 59, 56).unwrap(),
+                },
+                Interval {
+                    from_datetime: create_datetime(2021, 01, 01, 23, 59, 56).unwrap(),
+                    to_datetime: create_datetime(2021, 01, 02, 00, 00, 02).unwrap(),
+                },
+                Interval {
+                    from_datetime: create_datetime(2021, 01, 02, 00, 00, 02).unwrap(),
+                    to_datetime: create_datetime(2021, 01, 02, 00, 00, 08).unwrap(),
+                },
+                Interval {
+                    from_datetime: create_datetime(2021, 01, 02, 00, 00, 08).unwrap(),
+                    to_datetime: create_datetime(2021, 01, 02, 00, 00, 14).unwrap(),
+                },
+                Interval {
+                    from_datetime: create_datetime(2021, 01, 02, 00, 00, 14).unwrap(),
+                    to_datetime: create_datetime(2021, 01, 02, 00, 00, 20).unwrap(),
+                }
+            ]
+        );
+    }
+
+    #[actix_rt::test]
+    async fn temperature_charts() {
```

Comment:
test here check for aggregation of temperature logs for two sensor and one location